uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Squash.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    public Optional<Void> execute() {
+
+        this.isBare( repo );
+
+        final Git git = new Git( repo );
+
+        // resolve the objectId of the start commit. I will squash from this commit until the HEAD
+        ObjectId startCommitObjectId = this.getStartCommit( git, startCommitString );
+
+        RevWalk revWalk = new RevWalk( repo );
+        RevCommit startCommit = getRevCommit( startCommitObjectId, revWalk );
+        RevCommit parent = startCommit;
+        if ( startCommit.getParentCount() > 0 ) {
+            parent = getRevCommit( startCommitObjectId, revWalk ).getParent( 0 );
+        }
+
+        Ref head = this.getHead( repo );
+        this.markStart( revWalk, parent, head );
+
+        // Let's revert the order of the commits so they are sorted by commit time
+        revWalk.sort( RevSort.REVERSE );
+
+        PersonIdent commitAuthor = null;
+
+        Map<String, ObjectId> content = new HashMap<String, ObjectId>();
+        // Let's walk through each commit to collect author and changes
+        for ( RevCommit commit : revWalk ) {
+            //Should I check that all the commit authors are the same?
+            commitAuthor = commit.getAuthorIdent();
+            // let's walk to this commit tree to collect the path - objectId
+            content = collectPathAndObjectIdFromTree( this.repo, revWalk, commit );
+        }
+
+        revWalk.dispose();
+
+        final ObjectInserter odi = repo.newObjectInserter();
+
+        final ObjectId indexTreeId = createTemporaryIndex( git, content, odi );
+
+        // Create a commit object
+        final CommitBuilder commit = createCommit( parent, commitAuthor, indexTreeId, this.squashedCommitMessage );
+
+        // Insert the commit into the repository
+        final ObjectId commitId = insertCommitIntoRepositoryAndFlush( odi, commit );
+
+        updateReferenceAndReleaseRevisionWalk( git, revWalk, commitId );
+
+        return Optional.empty();
+    }
+
+    /**
+     * Create temporary index for commit content
+     * @param git
+     * @param content
+     * @param odi
+     * @return
+     * @throws {@link GitException} he cannot create temporary Index
+     */
+    private ObjectId createTemporaryIndex( Git git,
+                                           Map<String, ObjectId> content,
+                                           ObjectInserter odi ) {
+        try {
+            DirCache index = JGitUtil.createTemporaryIndexForContent( git, content );
+            return index.writeTree( odi );
+        } catch ( IOException e ) {
+            String message = "Cannot create temporary index form content";
+            throw new GitException( message, e );
+        }
+    }
+
+    /**
+     * @param git
+     * @param revWalk
+     * @param commitId
+     * @throws {@link ConcurrentRefUpdateException} if cannot lock head
+     * @throws {@link JGitInternalException} if updating ref failed
+     * @throws {@link IOException}
+     */
```

Comment:
we don't usually  javadoc private methods, unless super needed. and if needed, do for all